### ISSUE TITLE
FPPD dies with Sig 11 logging channel data (master branch only)

### ISSUE THREAD

**Issue by jfeger77** [FalconChristmas/fpp#923]
FPP Version 4.5 (Main release).
BBB K16 as master
BBB K32 as remote
Falcon Remote Receiver off of K32, and another off of K16
(Ethernet, no wifi anywhere)
VAST FMT direct off of BBB in K16 USB
VAST FPP Plugin
Big Buttons Plugin

Symptoms:
Sequence begins to play, shortly into the song (~45-50 seconds), FPP interface timers (Elapsed) stop, audio drops entirely, pixels still seem to be receiving data.  Seconds later Elapsed updates on the UI, audio comes back in (without missed lyrics), then drops, jitters, and drops.  Then the pixels began to drop and entire show goes dark.  FPPD restarts automatically, playlist kicks off as normal.

I did a quick replacement of the mp3 with one I bought from Apple (original from Amazon).  Same symptom and crash.

The sequence plays flawlessly in xlights 2020.54.
Im attaching the sequence and logs.  The base sequence came from the xLights share drive.  We added timing / lyric tracks and made some changes.

All other sequences and show working perfect.

[FPP-Logs.zip](https://github.com/FalconChristmas/fpp/files/5686317/FPP-Logs.zip)
[Sequence Package and RGB Effects.zip](https://github.com/FalconChristmas/fpp/files/5686321/Sequence.Package.and.RGB.Effects.zip)

**Comment by jfeger77** [comment]
Summary: DEBUG / ALL was accidentally left enabled, and this was causing the crash. I dont know why on this sequence and no others.  I restored INFO / MOST for logging and all is well.

......Longer version......

I did some testing to troubleshoot.  I fired up my spare K32, used FPP Backup to copy the channel output configuration to the test unit.  The sequence played fine on the test unit.  This told me 4.5 was capable of playing the sequence, and it must be a config issue.  I then used FPP backup to grab a full clone of my production Master, put it onto the test unit, and I was able to reproduce the crash in the exact spot.

I then started checking every section of config, and I noticed I left DEBUG / ALL enabled in logging, which was set for a previous issue.  I don't think this should cause a crash, but by setting it back to INFO / MOST solved the issue.

**Comment by jfeger77** [comment]
Reopening to make sure the DEV team sees this.

**Comment by ghormann** [comment]
You had Channel data logging enabled as well per the logs.  Honestly, I'm surprised the BBB could keep up.   :-)

To me, it looks like the  error is actually a signal 11 when writing the channel log. 

```
Thread 14 (Thread 0xaa5fbeb0 (LWP 4512)):
#0  __libc_do_syscall () at ../sysdeps/unix/sysv/linux/arm/libc-do-syscall.S:46
#1  0xb6714f7a in __waitpid (options=0, stat_loc=0xaa5fa404, pid=4558) at ../sysdeps/unix/sysv/linux/waitpid.c:30
#2  __waitpid (pid=4558, stat_loc=0xaa5fa404, options=0) at ../sysdeps/unix/sysv/linux/waitpid.c:25
#3  0x004377ac in handleCrash(int) [clone .part.52] ()
#4  <signal handler called>
#5  0xb4d05e7e in __wcslen (s=s@entry=0x80 <error: Cannot access memory at address 0x80>) at wcslen.c:31
#6  0xb4d06aa4 in __wcsrtombs (dst=0x0, src=0xaa5fafc8, src@entry=0x0, len=0, ps=ps@entry=0xaa5fafdc) at wcsrtombs.c:68
#7  0xb4cdb37c in _IO_vfprintf_internal (s=0xaeaf9e58, format=0xaa5fb478 "2020-12-13 19:57:31.314 (4512) common.cpp:195: 000000: 00 00 00 00 00 00 00 00 00 00 00 00 25 25 25 53    ", '.' <repeats 12 times>, "%%%S\n", ap=...) at vfprintf.c:1638
#8  0xb653795c in _LogWrite(char const*, int, int, int, char const*, ...) () from /opt/fpp/src/libfpp.so
#9  0xb651c566 in HexDump(char const*, void const*, int) () from /opt/fpp/src/libfpp.so
#10 0xb65032b2 in SendChannelData(char const*) () from /opt/fpp/src/libfpp.so
#11 0xb65050be in RunChannelOutputThread(void*) () from /opt/fpp/src/libfpp.so
#12 0xb670cbbe in start_thread (arg=0x9a949137) at pthread_create.c:486
#13 0xb4d4062c in ?? () at ../sysdeps/unix/sysv/linux/arm/clone.S:73 from /lib/arm-linux-gnueabihf/libc.so.6
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```

**Comment by jfeger77** [comment]
Heh, yeah I think I had it smoking.  The interesting thing was I noticed the temps were ~30-40 degrees C.  I chalked it up to my extra PSU in the enclosure.  Now ... ~12-15 degrees C.

Im not familiar with Signal 11.

**Comment by darylc** [comment]
Came across another end-user with this problem today.  I feel like we should put a red text alert on the front screen if logging channel data is enabled

**Comment by jfeger77** [comment]
> Came across another end-user with this problem today. I feel like we should put a red text alert on the front screen if logging channel data is enabled

Thats a good idea.  I was definitely in this mode of operation by accident.  I cant think of a reason to run in this mode intentionally during a normal show.

**Comment by ghormann** [comment]
The warning has already been added to the main screen of fpp via da81975f1eb270cc63def06b45bf126bdc193a74.   I believe this will be included in 4.6. 

![Warning Image](https://user-images.githubusercontent.com/24968726/100776084-1695d900-33d2-11eb-83b9-f97130a90e0d.png)

**Comment by darylc** [comment]
@ghormann The case I saw today was level info, but channel data, which isn't covered by that commit.

Resulted in fppd crashing with sig11

**Comment by ghormann** [comment]
Helpful. Updating title.

**Comment by ghormann** [comment]
Possible fix in 28f649c54ad87a02f0cd2ed068c9fe304a4f36eb which was part of 4.6

**Comment by cpinkham** [comment]
We also have plans to redo the logging UI to get rid of the 'ALL' option and provide for different log levels for each of the various logging categories/areas.  The intended plan is to allow you to do things like set Playlist and Scheduler to Debug while leaving Sequence and Channel Output at Warn and Channel Data at None.

**Comment by ghormann** [comment]
@jfeger77 where you able to verify this this solved your problem?

**Comment by ghormann** [comment]
Believe this was fixed in 28f649c which was part of 4.6.  No response from original poster to confirm.  Closing Issue for now.

**Comment by jfeger77** [comment]
@ghormann Sorry, I was away and didnt see this.  I did not upgrade off of 4.5, but the logging changes back to INFO / MOST did let my show run stable.
